### PR TITLE
TG-16 Employee model

### PR DESCRIPTION
**Pull request by SebastianRonchetti**

### CODE REVIEW

File: src/models/employee.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import mongoose from 'mongoose';
+
+const Schem = mongoose.Schema;
```
Comment (JuanMornik) [inline src/models/employee.js:3]: A mi me tira error el eslint si lo pongo así. Me obliga a ponerlo { Schem } = mongoose. Yo te recomiendo que te fijes si tenés habilitada la extensión de eslint

File: src/models/employee.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import mongoose from 'mongoose';
```
Comment (JuanMornik) [inline src/models/employee.js:1]: Te dejo el comentario acá porque en el nombre no me deja. Fijate que en la problemática pide que el nombre del archivo empiece con mayúscula, por ende sería: 'Employee.js'

**Review by JuanMornik — CHANGES_REQUESTED** [review]

File: src/models/employee.js (hunk 1 above)
Comment (SebastianRonchetti) [reply src/models/employee.js]: A mi me tiró la misma corrección pero cuando le puse Schem en vez de Schema no me la tiró más, pero si sigue estando la pongo como decís nomás.

File: src/models/EmployeeModel.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+const { Schema } = mongoose;
+
+const employeeSchema = new Schema(
+  {
+    fullName: {
+      type: String,
+      required: true,
+      min: 3,
+    },
```
Comment (AlfonsoDalix) [inline src/models/EmployeeModel.js:11]: I'd change it for first name, and last name

**Review by AlfonsoDalix — CHANGES_REQUESTED** [review]

File: src/models/EmployeeModel.js (hunk 3 above)
Comment (SebastianRonchetti) [reply src/models/EmployeeModel.js:11]: Done, also changed min > minlenght since min was for numeric values

**Review by Javier-Cervera — COMMENTED** [review]
I think there's no need to change file names. Such as:
employeeControllers --> employee
EmployeeModel --> model
Because the paths are self-referential

File: .gitignore [hunk 4]
```diff
@@ -104,3 +104,8 @@ dist
 .tern-port
 
 .DS_Store
+
+./src/controllers
+./src/routes
+./src/validations
+./index
```
Comment (AlfonsoDalix) [inline .gitignore:111]: Do not change any lines in the .gitignore file

File: src/models/employee.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    email: {
+      type: String,
+      required: true,
+      lowercase: true,
+    },
+    gender: {
+      type: String,
+      required: false,
+    },
```
Comment (AlfonsoDalix) [inline src/models/employee.js:25]: We could add an enum here like enum:['male', 'female', 'other']

File: src/models/EmployeeModel.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    active: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  { timestamps: true },
+);
+
+export default mongoose.model('Employee', employeeSchema);
```
Comment (AlfonsoDalix) [inline src/models/EmployeeModel.js:47]: If we have other document with the same name and lines, delete this file

File: src/controllers/employeeControllers.js [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+      if (err) {
+        res.send(err);
+      } else {
+        res.send('Employeed deleted');
+      }
+    });
+  }
+});
+
+export default router;
```
Comment (AlfonsoDalix) [inline src/controllers/employeeControllers.js:134]: We have to do it in a separate ticket

**Review by AlfonsoDalix — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (SebastianRonchetti) [reply .gitignore:111]: I had started working on the next ticket and added them to ignore the recent changes without having to delete everything

File: src/models/EmployeeModel.js (hunk 6 above)
Comment (SebastianRonchetti) [reply src/models/EmployeeModel.js:47]: Some mixup happened with the other branch I was woking on

File: src/controllers/employeeControllers.js (hunk 7 above)
Comment (SebastianRonchetti) [reply src/controllers/employeeControllers.js:134]: Yes, some mixup happened with the git console as I was already working on a new ticket, some errors were thrown and I thought I solved it. 
Evidently I was wrong

File: src/models/employee.js (hunk 5 above)
Comment (SebastianRonchetti) [reply src/models/employee.js]: Done

**Review by AlfonsoDalix — APPROVED** [review]

**Pull request merged**